### ISSUE TITLE
Looks like your test for 1.0.0 all failed on travis

### ISSUE THREAD

**Issue by binduwavell** [jonschlinkert/window-size#11]
All of the tests show green, but if you look at the details they say no test was specified. When I upgraded to 1.0.0 my tests on travis failed. I rolled back and now things are fine.

The error stack I was getting on travis looked like this:

```
/home/travis/build/binduwavell/generator-alfresco-common/node_modules/define-property/index.js:14
    throw new TypeError('expected an object or function.');
    ^
TypeError: expected an object or function.
    at defineProperty (/home/travis/build/binduwavell/generator-alfresco-common/node_modules/define-property/index.js:14:11)
    at Object.<anonymous> (/home/travis/build/binduwavell/generator-alfresco-common/node_modules/window-size/index.js:125:1)
...
```

**Comment by doowb** [comment]
@binduwavell I just updated the `package.json` to include the test script and it looks like all of the linux tests have passed so far (mac tests still running).

Are you able to run your code outside of travis? Is there anything else about your build or travis config that might be different?

**Comment by binduwavell** [comment]
Tests ran without issue on my mac. Which is why it took me ages to figure out it was window-size breaking travis.

I don't think there is anything strange about my build see binduwavell/generator-alfresco-common. I'll give it another try. What I do know is that reverting to window-size:0.3.0 resolved the issue.

**Comment by binduwavell** [comment]
I have made some progress. My tests are run by a js file called `ci.js` via the following entry in my `.travis.yml`:

```
script:
  - node ci.js
```

If I run `npm test` or even `npm run test:color` which is what `ci.js` mostly uses (one version is run with codecov) they work perfectly. However if I attempt to execute the tests via `./ci.js 6` (for example), the script failure happens.

Again, note that with 0.3.0 this worked just fine.

**Comment by binduwavell** [comment]
The following rather convoluted example in a directory with `window-size` installed reproduces the issue:

```
node -e "console.log(require('child_process').execSync(\"node -e \\\"console.log(require('window-size'))\\\"\"))"
```

Basically the following code works:

```
node -e "console.log(require('window-size'))"
```

However if you have `child_process.execSync()` make a new shell to run that same call we end up with the problem I'm seeing.

I did some investigation and it seems that the stdout stream is being used to grab the size.

As such, this works just fine on my mac:

```
node -e "console.log(require('child_process').execSync(\"node -e \\\"console.log(require('window-size'))\\\"\",{stdio:[process.stdout,process.stderr]}))"
```
I'll see if I can use this info to get things to work in travis.

**Comment by binduwavell** [comment]
I needed ci.js for older versions of node, but since I'm not supporting < 4 anymore I was able to get rid of it and simplify my build some. I still ran into the issue where size (width and height) come back as 0 on travis. I updated my code to have a default width (I don't care about height) and that is good enough for me for now.

**Comment by jonschlinkert** [comment]
@binduwavell thanks for providing more detail to help us work through this. these terminal sizing issues can be difficult to figure out. @doowb and I are still reviewing/catching up. let us know if you figure anything out, and we'll do the same on our end

**Comment by binduwavell** [comment]
I worked around the issue by having npm run the tests rather than the ci.js script and handling the 0 width with a fallback to 80. I don't plan to spend any more time trying to get window-size to return non 0 results on Travis. But if you figure it out I will happily test :)

**Comment by doowb** [comment]
@binduwavell I published `window-size@1.1.0` which I think will solve these issues you were having.

I'm going to close this issue, but if you're running into the same problem, please re-open and let us know.
